test/libsolidity/smtCheckerTests/array_members/push_as_lhs_3d.sol

```diff
@@ -0,0 +1,34 @@
+pragma experimental SMTChecker;
+
+contract C {
+	uint[][][] c;
+
+	function f() public {
+		require(c.length == 0);
+		c.push().push().push() = 2;
+		assert(c.length == 1);
+		assert(c[0].length == 1);
+		assert(c[0][0].length == 1);
+		assert(c[0][0][0] == 2);
+	}
+
+	function g() public {
+		c.push().push().push() = 2;
+		uint length1 = c.length;
+		uint length2 = c[length1 - 1].length;
+		uint length3 = c[length1 - 1][length2 - 1].length;
+		assert(length1 > 0);
+		assert(length2 == 1);
+		assert(length3 == 1);
+		assert(c[length1 - 1][length2 - 1][length3 - 1] == 2);
+		// Fails
+		assert(c[length1 - 1][length2 - 1][length3 - 1] == 200);
+	}
+}
+// ----
+// Warning 6328: (431-450): CHC: Assertion violation might happen here.
+// Warning 6328: (454-474): CHC: Assertion violation might happen here.
+// Warning 6328: (478-498): CHC: Assertion violation might happen here.
+// Warning 6328: (502-555): CHC: Assertion violation might happen here.
+// Warning 6328: (570-625): CHC: Assertion violation might happen here.
```

Comment:
Why CHC reports that violation might happen?

Reply:
Both engines report `might happen` when they're not able to solve a query.

Reply:
Maybe we should actually change that message to something more precise

Reply:
We should consider changing that message probably. We should open issue for that and do it sometime in future.